src/bundle/Resources/config/services/helpers.yaml

```diff
@@ -9,6 +9,7 @@ services:
 
     EzSystems\EzRecommendationClient\Helper\SiteAccessHelper:
         arguments:
+            $siteAccess: '@ezpublish.siteaccess'
```

Comment:
SiteAccess should be fetched via `\eZ\Publish\Core\MVC\Symfony\SiteAccess\SiteAccessServiceInterface` (`@ezpublish.siteaccess_service`).